src/hooks/useChains.ts

```diff
@@ -25,3 +26,14 @@ export const useCurrentChain = (): ChainInfo | undefined => {
   const chainInfo = useAppSelector((state) => selectChainById(state, chainId), isEqual)
   return chainInfo
 }
+
+/**
+ * Checks if a feature is enabled on the current chain.
+ *
+ * @param feature name of the feature to check for
+ * @returns `true`, if the feature is enabled on the current chain. Otherwise `false`
+ */
+export const useHasFeature = (feature: FEATURES): boolean => {
```

Comment:
We still have a few places where `hasFeature` is used. Is this intended or should we update those to the hook as well?

Reply:
For now I updated all the components / hook which did not need the `chainInfo` at all but were using the pattern of 

```ts
const chainInfo = useCurrentChain()
const isEnabled = hasFeature(chainInfo, featureName)
```

some hooks actually need the chainInfo as well. Those ones I did not touch.
Sometimes `hasFeature` is also used outside of hooks / components.